docs/guides/postgres/backup/kubestash/customization/index.md

```diff
@@ -0,0 +1,414 @@
+          - name: logical-backup
+```
+
+> You can configure additional runtime settings for backup jobs within the `addon.jobTemplate.spec` sections. For further details, please refer to the [reference](https://kubestash.com/docs/latest/concepts/crds/backupconfiguration/#podtemplate-spec).
+
+## Customizing Restore Process
+
+`KubeStash` uses [psql](https://www.postgresql.org/docs/current/app-psql.html) during the restore process. In this section, we are going to show how you can pass arguments to the restore process, restore a specific snapshot, run restore job as a specific user, etc.
+
+### Passing arguments to the restore process
+
+You can pass any supported `psql` arguments to the restore process using the `args` field within the `addon.tasks[*].params` section. This example demonstrates how to specify a database `testdb` to connect to during the restore process.
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: RestoreSession
+metadata:
+  name: sample-postgres-restore
+  namespace: demo
+spec:
+  target:
+    apiGroup: kubedb.com
+    kind: Postgres
+    namespace: demo
+    name: restored-postgres
+  dataSource:
+    repository: gcs-postgres-repo
+    snapshot: latest
+    encryptionSecret:
+      name: encrypt-secret
+      namespace: demo
+  addon:
+    name: postgres-addon
+    tasks:
+      - name: logical-backup-restore
+        params:
+          args: --dbname=testdb
+```
+
+### Restore specific snapshot
+
+You can also restore a specific snapshot. At first, list the available snapshot as bellow,
+
+```bash
+$ kubectl get snapshots.storage.kubestash.com -n demo -l=kubestash.com/repo-name=gcs-postgres-repo
+NAME                                                                    REPOSITORY          SESSION           SNAPSHOT-TIME          DELETION-POLICY   PHASE       AGE
+gcs-postgres-repo-sample-postgres-backup-frequent-backup-1725257849   gcs-postgres-repo   frequent-backup   2024-09-02T06:18:01Z      Delete            Succeeded   15m
+gcs-postgres-repo-sample-postgres-backup-frequent-backup-1725258000   gcs-postgres-repo   frequent-backup   2024-09-02T06:20:00Z      Delete            Succeeded   13m
+gcs-postgres-repo-sample-postgres-backup-frequent-backup-1725258300   gcs-postgres-repo   frequent-backup   2024-09-02T06:25:00Z      Delete            Succeeded   8m34s
+gcs-postgres-repo-sample-postgres-backup-frequent-backup-1725258600   gcs-postgres-repo   frequent-backup   2024-09-02T06:30:00Z      Delete            Succeeded   3m34s
+```
+
+The below example shows how you can pass a specific snapshot name in `.spec.dataSource` section.
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: RestoreSession
+metadata:
+  name: sample-postgres-restore
+  namespace: demo
+spec:
+  target:
+    apiGroup: kubedb.com
+    kind: Postgres
+    namespace: demo
+    name: restored-postgres
+  dataSource:
+    repository: gcs-postgres-repo
+    snapshot: gcs-postgres-repo-sample-postgres-backup-frequent-backup-1725258000
```

Comment:
This is necessary to provide the repo name to restore a specific snapshot?

Reply:
Yes! As far as I know, it's a required field.
https://github.com/kubestash/apimachinery/blob/master/apis/core/v1alpha1/restoresession_types.go#L221